android/build.gradle

```diff
@@ -1,22 +1,23 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.1.3'
+        classpath 'com.android.tools.build:gradle:3.2.1'
     }
 }
 
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 26
-    buildToolsVersion "26.0.1"
+    compileSdkVersion rootProject.ext.compileSdkVersion
```

Comment:
If the rootProject doesn't specify that, which in rare cases, the build would fail. And I find [this template](https://raw.githubusercontent.com/itinance/react-native-fs/master/android/build.gradle) from react-native-fs could be used at here.

Reply:
root will be necessary defined in the main build.gradle if you use standard template defined by react-native.
With 0.58 migration, main build.gradle migration is required since gradle tool is also updated with gradle-wrapper.properties (gradle-4.7-all.zip)

I advice that you try default react-native module  template:
```
react-native init test-app
```
and try to link react-native-zip-archive

official react-native module template is not up to date :( and cannot help)
https://facebook.github.io/react-native/docs/native-modules-setup

Do not hesitate to remove this pull request when you will fix this problem.